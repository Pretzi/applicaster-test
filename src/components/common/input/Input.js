import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Input extends Component {
  render() {
    const {
      value,
      onChange,
      placeholder,
      type,
      name,
      icon
    } = this.props;

    return (
      <div className="pt-input-group pt-large">
        <span className={`pt-icon ${icon}`} />
        <input
          className="pt-input pt-large"
          type={type}
          placeholder={placeholder}
          dir="auto"
          value={value}
          onChange={(e) => onChange(e)}
          name={name}
        />
      </div>
    );
  }
}

Input.displayName = 'App.Input';
Input.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  icon: PropTypes.string
};

export default Input;