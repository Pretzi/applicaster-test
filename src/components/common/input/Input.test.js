import React from 'react';
import Input from './Input';
import { shallow } from 'enzyme';

describe('<Input />', () => {
  it('Should render as expected', () => {
    const wrapper = shallow(
      <Input 
        value="test" 
        placeholder="Search" 
        type="search" 
        onChange={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  })
});