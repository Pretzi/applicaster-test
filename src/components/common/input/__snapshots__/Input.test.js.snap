// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App.Input
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="pt-icon undefined"
/>,
        <input
          className="pt-input pt-large"
          dir="auto"
          name={undefined}
          onChange={[Function]}
          placeholder="Search"
          type="search"
          value="test"
/>,
      ],
      "className": "pt-input-group pt-large",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "pt-icon undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "pt-input pt-large",
          "dir": "auto",
          "name": undefined,
          "onChange": [Function],
          "placeholder": "Search",
          "type": "search",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="pt-icon undefined"
/>,
          <input
            className="pt-input pt-large"
            dir="auto"
            name={undefined}
            onChange={[Function]}
            placeholder="Search"
            type="search"
            value="test"
/>,
        ],
        "className": "pt-input-group pt-large",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "pt-icon undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "pt-input pt-large",
            "dir": "auto",
            "name": undefined,
            "onChange": [Function],
            "placeholder": "Search",
            "type": "search",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
